controllers/component_controller.go

```diff
@@ -434,8 +474,10 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 				return ctrl.Result{}, err
 			}
 
-			// Generate and push the gitops resources, if necessary.
 			component.Status.ContainerImage = component.Spec.ContainerImage
+			component.Status.Devfile = string(yamlHASCompData)
```

Comment:
Can you call `r.Status().Update()` after here?

Reply:
Do we need to? Because I checked and made sure that our `func SetGitOpsGeneratedConditionAndUpdateCR` calls `err := r.Client.Status().Update(ctx, component)` irrespective of an err or not 🤔

Reply:
It seems like the next steps are either generateGitops or SetUpdateConditionAndUpdateCR which both call the `r.Client.Status().Update()`

Reply:
It’s for the gitops job, it needs to Kubectl get the component since we don’t have access to it directly

Reply:
If the tests don’t pass with the change - don’t worry.

Reply:
Understood, updated the PR!